state.go

```diff
@@ -557,6 +561,103 @@ func (s *State) Channel(channelID string) (*Channel, error) {
 	return nil, ErrStateNotFound
 }
 
+// ThreadMemberUpdate updates the ThreadMember object for the user, in a specific thread.
+func (s *State) ThreadMemberUpdate(t *ThreadMember) error {
+	if s == nil {
+		return ErrNilState
+	}
+
+	// If the channel exists in the state, update the thread member of that channel
+	if channel, ok := s.channelMap[t.ID]; ok {
+		s.Lock()
+		defer s.Unlock()
+
+		channel.Member = t
+		return nil
+	}
+
+	return ErrStateNotFound
+}
```

Comment:
This is copied from https://github.com/bwmarrin/discordgo/pull/922/commits/d890873d60f2958fa010609793c483ed2d0ab59e#diff-320959764bfa2277e0bdd4eaa741f02891d123c42150d4ccb76de0bdbf6e591aR563-R580, and includes the same bug of the unsynchronized access of `s.channelMap`.